### PR TITLE
Buildah dev plugin 1.14.0

### PR DESCRIPTION
**Pull request by vitaliy-guliy**
### What does this PR do?

A plugin which adds `buildah-dev` container to the workspace to build docker images.

At the moment, it includes pretty old but stable buildah v1.14.0.
Anyway, it's stable and works pretty good.

Before merging we need to:
- [x] merge PR for the buildah-base image v 1.14.0 https://github.com/che-dockerfiles/che-buildah-base/pull/1
- [x] build and push the docker image on quay.io e.g. `quay.io/eclipse/che-buildah-base:1.14.0`
- [x] change the reference to the docker image. 

Tested with devfile
```
# Copyright (c) 2020 Red Hat, Inc.
# This program and the accompanying materials are made
# available under the terms of the Eclipse Public License 2.0
# which is available at https://www.eclipse.org/legal/epl-2.0/
#
# SPDX-License-Identifier: EPL-2.0
#
# See: https://sipb.mit.edu/doc/safe-shell/

apiVersion: 1.0.0

metadata:
  name: kubernetes-tooling-dev

projects:
  - name: che-sidecar-kubernetes-tooling
    source:
      location: 'https://github.com/che-dockerfiles/che-sidecar-kubernetes-tooling'
      type: git
      branch: '1.1.0'

components:
  - type: cheEditor
    id: eclipse/che-theia/7.10.0
    alias: che-theia
    memoryLimit: 3072Mi

  - type: dockerimage
    image: 'quay.io/eclipse/che-theia-dev:next'
    alias: che-dev
    mountSources: true
    endpoints:
      - name: theia-dev-flow
        port: 3010
        attributes:
          protocol: http
          public: 'true'
    memoryLimit: 2048Mi

  - type: chePlugin
    reference: >-
      https://raw.githubusercontent.com/vitaliy-guliy/che-plugin-registry/buildah-base/v3/plugins/containers/buildah/1.14.0/meta.yaml
    memoryLimit: 3Gi
    alias: buildah-dev

commands:

  #
  # Logins to quay.io
  #
  - name: '3.1 Login to quay.io'
    actions:
      - workdir: /projects/che-sidecar-kubernetes-tooling
        type: exec
        command: |
          if [ ! -f "/tmp/username" ]; then
            read -p "Username: " USERNAME &&
            buildah login --username $USERNAME quay.io &&
            echo $USERNAME > /tmp/username &&
            echo -e "\e[32mDone.\e[0m"
          else
            echo 'You already logged in as' `cat /tmp/username`
          fi
        component: buildah-dev

  #
  # Builds image with Kubernetes plugin
  #
  - name: '3.2 Build the image'
    actions:
      - workdir: /projects/che-sidecar-kubernetes-tooling
        type: exec
        command: |
          if [ -f "/tmp/username" ]; then
              USERNAME=`cat /tmp/username`
              buildah bud -t quay.io/$USERNAME/che-sidecar-kubernetes-tooling:dev .
              echo 'Image built'
              buildah images
              echo -e "\e[32mDone.\e[0m"
          else
              echo 'You need to login the first'
          fi
        component: buildah-dev

  #
  # Pushes the image to quay.io
  # 
  - name: '3.3 Push to quay.io'
    actions:
      - workdir: /projects/che-sidecar-kubernetes-tooling
        type: exec
        command: |
          if [ -f "/tmp/username" ]; then
              USERNAME=`cat /tmp/username`
              PLUGIN_IMAGE_ID=`buildah images -q quay.io/$USERNAME/che-sidecar-kubernetes-tooling`
              echo 'The last built image ID is :' $PLUGIN_IMAGE_ID
              PLUGIN_IMAGE=quay.io/$USERNAME/che-sidecar-kubernetes-tooling:dev
              echo 'Pushing image' $PLUGIN_IMAGE
              echo $PLUGIN_IMAGE > /tmp/plugin_image
              buildah push $PLUGIN_IMAGE_ID docker://$PLUGIN_IMAGE
              echo -e "\e[32mDone.\e[0m"
          else
              echo 'The first, you need to login and build the image'
          fi
        component: buildah-dev
```

### CODE REVIEW

File: v3/plugins/containers/buildah/1.14.0/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+title: Buildah Base Dev Plugin
+description: Container to build docker images with buildah
+icon: https://raw.githubusercontent.com/containers/buildah/v1.14.0/logos/buildah-logomark_large.png
+repository: https://github.com/containers/buildah
+firstPublicationDate: "2020-04-15"
+category: Other
+spec:
+  containers:
+   - name: buildah-dev
+     image: "quay.io/vgulyy/che-buildah-base:1.14.0"
```
Comment (ericwill) [inline v3/plugins/containers/buildah/1.14.0/meta.yaml:16]: If you go into the quay.io repo for che-buildah-base, you can see a successful image built from your PR to the che-dockerfiles/che-buildah-base image. In this case it should be something like `quay.io/eclipse/che-buildah-base:1.14.0-b0c91d3`

**Comment by vitaliy-guliy** [comment]
Have just tested with this image `quay.io/eclipse/che-buildah-base:1.14.0-b0c91d3`
Building and pushing the image works fine.

File: v3/plugins/containers/buildah/1.14.0/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+icon: https://raw.githubusercontent.com/containers/buildah/v1.14.0/logos/buildah-logomark_large.png
+repository: https://github.com/containers/buildah
+firstPublicationDate: "2020-04-15"
+category: Other
+spec:
+  containers:
+   - name: buildah-dev
+     image: "quay.io/eclipse/che-buildah-base:1.14.0-b0c91d3"
+     mountSources: true
+     memoryLimit: "3Gi"
```
Comment (amisevsk) [inline v3/plugins/containers/buildah/1.14.0/meta.yaml:18]: Is `3Gi` really the requirement here? Setting this as a default could cause a lot of issues in adding it to a workspace.

File: v3/plugins/containers/buildah/1.14.0/meta.yaml (hunk 1 above)
Comment (vitaliy-guliy) [reply v3/plugins/containers/buildah/1.14.0/meta.yaml]: Done

File: v3/plugins/containers/buildah/1.14.0/meta.yaml (hunk 2 above)
Comment (vitaliy-guliy) [reply v3/plugins/containers/buildah/1.14.0/meta.yaml:18]: I totally agree with you. The value is changed to 1gb, build checked on image for kubernetes plugin sidecar.

**Comment by vitaliy-guliy** [comment]
@ericwill let's approve and merge this PR if there are no more feedbacks

**Pull request merged**